src/plugins/discover/public/application/main/utils/use_discover_state.ts

```diff
@@ -207,20 +157,15 @@ export function useDiscoverState({
     [refetch$, searchSessionManager]
   );
 
+  /**
+   * Update state on dataView or saved search
+   */
```

Comment:
Adding more comments FTW! I'd suggest to provide a bit more details here that is code is executed when an indexPattern/dataView was switched, which also happens when a savedSearch was loaded (even if the savedSearch has the same dataView) .